simple_graph.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+class Node(object):
+    """Node."""
+
+    def __init__(self, value):
+        """Init Node."""
+        self.value = value
+
+
+class Graph(object):
+    """The Graph."""
+
+    def __init__(self):
+        """Init Graph."""
+        self.container = {}
+
+    def add_node(self, value):
+        """Add node to Graph."""
+        self.container.setdefault(value, [])
+
+    def has_node(self, value):
+        """Check if node is in the graph."""
+        return value in self.container
+
+    def add_edge(self, pointer_node, destination_node):
+        """Add edge between two nodes. If not in graph add nodes and edge."""
+        self.container.setdefault(pointer_node, []).append(destination_node)
+
```

Comment:
This only adds the pointer node if it's not already in the graph. It doesn't add the destination node to the graph, it just adds it as a neighbor to the pointer node.